src/modules/stream-buffer.js

```diff
@@ -0,0 +1,61 @@
+const assert = require('assert');
+
+/*
+ *  @class StreamBuffer
+ *  Buffer for the stream.
+ *  Bufferizes subscribers and adapters for further auto connection to the main link
+ *  When initialising, stream looks into a buffer and pulls all the subscribers and adapters from here
+ */
+class StreamBuffer {
+  /*
+    @constructor
+    Construct StreamBuffer instance with the following properties:
+    adapters { object } - map of name -> adapter { function }
+    subscribers { array } - array of subscriber classes
+   */
+  constructor() {
+    this.adapters = {};
+    this.subscribers = [];
+  }
+  /**
+   * @function addSubscriber - adds subscriber to the buffer
+   * @param CustomSubscriber {function|object} - object or a function with the constructor of a subscriber
+   * If function - should be a constructor
+   * If object - should contain config object and constructor class
+   */
+  addSubscriber(CustomSubscriber) {
```

Comment:
This one is only used by chain links init function to add it to buffer so that a logger will then automatically apply them on every reconfiguration.

Basically this is called when for example doing ``require('logtify-logstash')()``. You only call it once and I don't think there is a need to let one chain link remove something from the buffer. It is more related to user. If you want to use a chain link - you require it and if you don't - then do not require it.